src/sinks/aws_cloudwatch_metrics/mod.rs

```diff
@@ -139,7 +138,7 @@ impl SinkConfig for CloudWatchMetricsSinkConfig {
 
 impl CloudWatchMetricsSinkConfig {
     async fn healthcheck(self, client: CloudwatchClient) -> crate::Result<()> {
-        let _result = client
+        let _ = client
```

Comment:
It isn't letting me select the whole body of the function to suggest change , but - 

I liked that suggested approach discussed offline to use `.map()` to map the Ok(stuff) -> Ok(()) , and not have the variable. Seems a much cleaner solution.

Reply:
🤔 is this any different from the `map` option?

```rust
    async fn healthcheck(self, client: CloudwatchClient) -> crate::Result<()> {
        client
            .put_metric_data()
            .metric_data(
                MetricDatum::builder()
                    .metric_name("healthcheck")
                    .value(1.0)
                    .build(),
            )
            .namespace(&self.default_namespace)
            .send()
            .await?;

        Ok(())
    }
```

Reply:
Just did ☝️ in https://github.com/vectordotdev/vector/pull/14477/commits/7b67947cbcc7fbbc32b73eee2cdded908c8747c7 as it seemed less painful (to me). If everyone prefers some mapping to the error types then we can change it around still.